### PR TITLE
internal/v5: make DELETE work again

### PR DESCRIPTION
**Pull request by rogpeppe**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2078/

**Review by jrwren — COMMENTED** [review]
What about the resources attached to the deleted entity? Are they now orphaned?

**Comment by rogpeppe** [comment]
@jrwren resources are attached to the charm as a whole, not a specific revision within a charm, so they're not orphaned as such. But that raises the interesting question of whether we allow the deletion of the only (unpublished) revision of a charm when there are some resources. A bit more thought required there, I think.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2079/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2080/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2081/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2082/

**Review by fabricematrat — APPROVED** [review]

File: internal/charmstore/store_test.go [hunk 1]
```diff
@@ -4194,6 +4194,75 @@ func (s *StoreSuite) TestPublishWithFailedESInsert(c *gc.C) {
...
+	store := s.newStore(c, false)
+	defer store.Close()
+	url := router.MustNewResolvedURL("~charmers/precise/wordpress-12", -1)
+	err := store.AddCharmWithArchive(url, storetesting.NewCharm(&charm.Meta{
+		Series: []string{"precise"},
+	}))
+	c.Assert(err, gc.Equals, nil)
+
+	err = store.DeleteEntity(url)
+	c.Assert(err, gc.ErrorMatches, `cannot delete last revision of charm or bundle`)
```
Comment (frankban) [inline internal/charmstore/store_test.go:4238]: Check that the entity is actually still there?

**Review by frankban — APPROVED** [review]
Looks good! Thanks for restoring this functionality!

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**